### ISSUE TITLE
Support input files that do not end with ".pdf"

### ISSUE THREAD

**Issue by petervwyatt** [veraPDF/veraPDF-library#1099]
VeraPDF GUI and CLI both require all input to be named "*.pdf", however I cannot see any statement in any part of ISO 19005 which explicitly requires this. Furthermore, some cloud-based systems do not natively store files in their backend with file extensions (they use metadata) which makes using now veraPDF impossible.

Current veraPDF GUI does even provide an option in the file open dialog to match "All files (*.*)".
Current veraPDF CLI output:
```
Jul. 22, 2020 4:27:23 PM org.verapdf.apps.utils.ApplicationUtils filterPdfFiles
SEVERE: File xxxxx doesn't have a .pdf extension.
Jul. 22, 2020 4:27:23 PM org.verapdf.cli.VeraPdfCliProcessor processFilePaths
SEVERE: There are no files to process.
```

Expected behaviour is that any file can be checked for PDF/A compliance - or, at the very most, a warning is issued that the input file does not have ".pdf" file extension. It is certainly not a SEVERE error as indicated in the CLI output!

**Comment by bdoubrov** [comment]
I think this filter was introduced primarily for batch processing, where we skip all files with non-pdf extension. I do agree that at least for a single file processing all files should be accepted ignoring their file extension.

**Comment by petervwyatt** [comment]
Ideally an option to also permit batch processing (multi-file) for backend servers or corpora would be good. I understand if you don't want to make it default behaviour.

**Comment by bdoubrov** [comment]
Added optional CLI argument `--nonpdfext` to allow files with non-pdf extension. 

Also enabled "All files" file type filer in the GUI.

This functionality is available in the dev version veraPDF 1.17.4 and will be included into the next release: https://software.verapdf.org/dev/1.17/verapdf-greenfield-1.17.4-installer.zip
